controllers/tenantcontrolplane_controller.go

```diff
@@ -315,3 +316,17 @@ func (r *TenantControlPlaneReconciler) dataStore(ctx context.Context, tenantCont
 
 	return ds, nil
 }
+
+func (r *TenantControlPlaneReconciler) serviceAccount(ctx context.Context, tenantControlPlane *kamajiv1alpha1.TenantControlPlane) (*corev1.ServiceAccount, error) {
+    serviceAccountName := tenantControlPlane.Spec.ControlPlane.Deployment.ServiceAccountName
+    if len(serviceAccountName) == 0 {
+        serviceAccountName = "default"
+    }
+
+    sa := &corev1.ServiceAccount{}
+    if err := r.Client.Get(ctx, k8stypes.NamespacedName{Name: serviceAccountName, Namespace: tenantControlPlane.Namespace}, sa); err != nil {
+		return nil, errors.Wrap(err, "cannot retrieve serviceAccount")
+    }
+
+    return sa, nil
+}
```

Comment:
As discussed, this would increase the RBAC for Kamaji.
Kubernetes allows the creation of resources even with no ServiceAccount for the eventually consistent approach.
Must be removed.

Reply:
That is true. I must have forgotten it there since i had already started writing the code when we were discussing that point. Will remove it.